server/users.js

```diff
@@ -2,30 +2,117 @@
 
 const db = require('APP/db')
 const User = db.model('users')
+const Address = db.model('addresses')
 
 const {mustBeLoggedIn, forbidden} = require('./auth.filters')
+const router = require('express').Router()
 
-module.exports = require('express').Router()
-  .get('/',
-    // The forbidden middleware will fail *all* requests to list users.
-    // Remove it if you want to allow anyone to list all users on the site.
-    //
-    // If you want to only let admins list all the users, then you'll
-    // have to add a role column to the users table to support
-    // the concept of admin users.
-    forbidden('listing users is not allowed'),
-    (req, res, next) =>
-      User.findAll()
-        .then(users => res.json(users))
-        .catch(next))
-  .post('/',
-    (req, res, next) =>
-      User.create(req.body)
-      .then(user => res.status(201).json(user))
-      .catch(next))
-  .get('/:id',
-    mustBeLoggedIn,
-    (req, res, next) =>
-      User.findById(req.params.id)
-      .then(user => res.json(user))
-      .catch(next))
+module.exports = router
+
+router.param(':userId', (req, res, next, userId) => {
+  User.find({
+    where: {id: userId}
+  })
+  .then(targetUser => {
+    if (!targetUser) return res.sendStatus(404)
+    req.targetUser = targetUser
+  })
+  .catch(next)
+})
+
+router.param(':addressType', (req, res, next, addressType) => {
+  if (addressType !== 'BillingAddress' || addressType !== 'ShippingAddress') {
+    res.sendStatus(404).end()
+  }
+  req.addressType = addressType
+  next()
+})
+
+router.route('/')
+  // The forbidden middleware will fail *all* requests to list users.
+  // Remove it if you want to allow anyone to list all users on the site.
+  //
+  // If you want to only let admins list all the users, then you'll
+  // have to add a role column to the users table to support
+  // the concept of admin users.
+.get(forbidden('listing users is not allowed'), (req, res, next) => {
+  User.findAll()
+    .then(users => res.json(users))
+    .catch(next)
+})
+
+.post((req, res, next) => {
+  User.create(req.body)
+  .then(user => res.status(201).json(user))
+  .catch(next)
+})
+
+router.route('/:userId')
+
+.get(mustBeLoggedIn, (req, res, next) => {
+  res.json(req.targetUser)
+})
+
+.put(mustBeLoggedIn, (req, res, next) => {
+  req.targetUser.update(res.body)
+  .then(updatedUser => {
+    res.json(updatedUser)
+  })
+  .catch(next)
+})
+
+.delete(forbidden('must be admin'), (req, res, next) => {
+  req.targetUser.destroy()
+  .then(() => res.sendStatus(204))
+  .catch(next)
+})
+
+/*  -------------- ADDRESSES ------------- */
+
+router.route('/:userId/:addressType')
+
+.get(mustBeLoggedIn, (req, res, next) => {
+  req.targetUser['get' + req.addressMethod]()
```

Comment:
watch your variable names -- believe this needs to be req.addressType as defined in your router.param up top